wave_matching/src/ndt.cpp

```diff
@@ -0,0 +1,63 @@
+#include "wave/utils/config.hpp"
+#include "wave/utils/logging.hpp"
+#include "wave/matching/ndt.hpp"
+
+namespace wave {
+
+NDTMatcher::NDTMatcher(float res, const std::string &config_path) {
+    this->ref = boost::make_shared<pcl::PointCloud<pcl::PointXYZ> >();
+    this->target = boost::make_shared<pcl::PointCloud<pcl::PointXYZ> >();
+    this->final = boost::make_shared<pcl::PointCloud<pcl::PointXYZ> >();
+
+    ConfigParser parser;
+
+    float step_size = 0.1, t_eps = 0.01, config_res = 3;
+    int max_iter = 100;
+
+    parser.addParam("ndt.step_size", &step_size);
+    parser.addParam("ndt.max_iter", &max_iter);
+    parser.addParam("ndt.t_eps", &t_eps);
+    parser.addParam("ndt.config_res", &config_res);
+
+    if (parser.load(config_path) != 0) {
+        ConfigException config_exception;
+        throw config_exception;
+    }
+
+    if (res < this->min_res) {
+        if(config_res < this->min_res) {
+            LOG_ERROR("Invalid resolution given, using minimum");
```

Comment:
The general error logging macro, it is defined [here](https://github.com/wavelab/libwave/blob/master/wave_utils/include/wave/utils/logging.hpp)